src/wallet/walletdb.cpp

```diff
@@ -742,8 +745,8 @@ DBErrors WalletBatch::LoadWallet(CWallet* pwallet)
             if (!strErr.empty())
                 pwallet->WalletLogPrintf("%s\n", strErr);
         }
-        pcursor->close();
```

Comment:
In commit "walletdb: Handle cursor internally" (cda86544ae9de430e4558b87945046d5612202ac)

Minor: I think it'd be a little clearer to keep this close here as m_batch.CloseCursor() so it's more obvious cursor is closed in all cases, and so you don't need to add a new close call in a different place above.

Reply:
But above, in https://github.com/bitcoin/bitcoin/pull/19308/files#diff-e506682270036e4c8e0b8159a34b652dR723, it returns so it needs to be closes there, or am I missing something?

Reply:
I've moved it to after the `catch` block and removed the ones inside the loop except for the ones right before the early return.